addons/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/connector/PowermaxConnector.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.powermax.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.openhab.binding.powermax.internal.message.PowermaxBaseMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An abstract class for the communication with the Visonic alarm panel that
+ * handles stuff common to all communication types
+ *
+ * @author Laurent Garnier
+ * @since 1.9.0
+ */
+public abstract class PowermaxConnector implements PowermaxConnectorInterface {
+
+    private final Logger logger = LoggerFactory.getLogger(PowermaxConnector.class);
+
+    private InputStream input;
+    private OutputStream output;
+    private boolean connected = false;
+    private Thread readerThread;
```

Comment:
The thread is run continously.
I don't want to change that because they are some very subtil stuff with the serial connection. In the past, the serial port was not released correctly.
I am using a very similar pattern to the RFXCOM binding.

Reply:
👍